lib/eth/key/encrypter.rb

```diff
@@ -0,0 +1,191 @@
+# Copyright (c) 2016-2022 The Ruby-Eth Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'json'
+require 'securerandom'
+require 'openssl'
+require 'scrypt'
+
+# The Eth::Key::Encrypter class to handle PBKDF2-SHA-256 encryption
+class Eth::Key::Encrypter
+  include Eth::Utils
+
+  # Class method `Eth::Key::Encrypter.perform`
+  #
+  # @return [JSON] formatted with encrypted key (cyphertext) and [other identifying data](https://github.com/ethereum/wiki/wiki/Web3-Secret-Storage-Definition#pbkdf2-sha-256)
+  def self.perform key, password, options = {}
+    new(key, options).perform(password)
+  end
+
+  # Constructor of the `Eth::Key::Encrypter` class for secret key
+  # encryption.
+  #
+  # @param key [String] representing a secret key used for encryption
+  # @param options [Hash] the options to encrypt with
+  # @option options [String] :kdf key derivation function defaults to pbkdf2
+  # @option options [String] :skip_address [unecessary and compromises privacy](https://github.com/ethereum/wiki/wiki/Web3-Secret-Storage-Definition#alterations-from-version-1)
```

Comment:
let's omit addresses completely, they are cheap to generate later, so we can also omit the `skip_address` feature.